blockprotocol/blockprotocol
# Handle URL params in docs pages

There are a few places in the docs where matching on the URL path or anchor is relied on: to scroll to sections and highlight the current page/section in the navigation menus.

Including params currently causes docs pages to loop, as they assume a match on the page path will be found, but it won't. e.g. https://blockprotocol.org/docs?test=test

We want to send an email with UTM params, which this is blocking.

### Known issues

In the mobile nav menu, items below depth 2 are not highlighted, _when a URL param is present_. I didn't spend very long looking at this because it seems unlikely to occur.

### How to test

Visit various paths with a URL param and check:
1. Scroll to section works
2. Highlighting in the desktop nav on the left works
3. Highlighting on the mobile menu work
4. Check the breadcrumbs work (on medium / tablet screen sizes)
